ocaml/xapi/storage_impl.ml

```diff
@@ -113,9 +113,11 @@ let info (fmt : ('a, unit, string, unit) format4) =
 let host_state_path = ref "/var/run/nonpersistent/xapi/storage.db"
 
 module Dp = struct
-  type t = string [@@deriving rpcty]
+  type t = string * Vm.t [@@deriving rpcty]
 
-  let make username = username
+  let make ((_datapath, _vm) as dp) = dp
+
+  let to_string (dp, _) = dp
```

Comment:
Types already let us know that a string is returned, stating the semantic meaning  seems more valuable:
```suggestion
  let to_path (dp, _) = dp
```